packages/zoe/src/contracts/multipoolAutoswap/pool.js

```diff
@@ -0,0 +1,175 @@
+import { E } from '@agoric/eventual-send';
+import { assert, details } from '@agoric/assert';
+
+import {
+  getInputPrice,
+  calcLiqValueToMint,
+  calcValueToRemove,
+  trade,
+  assertUsesNatMath,
+} from '../../contractSupport';
+
+import '../../../exported';
+
+/**
+ * @param {ContractFacet} zcf
+ * @param {(brand: Brand) => boolean} isSecondary
+ * @param {(brand: Brand, pool: Pool) => void} initPool
+ * @param {Brand} centralBrand
+ */
+export const makeAddPool = (zcf, isSecondary, initPool, centralBrand) => {
+  const makePool = (liquidityZcfMint, poolSeat, secondaryBrand) => {
+    let liqTokenSupply = 0;
+    const {
+      brand: liquidityBrand,
+      amountMath: liquidityAmountMath,
+    } = liquidityZcfMint.getIssuerRecord();
+
+    /**
+     * @typedef {Object} Pool
+     * @property {(centralToSecondary: boolean, inputValue: Value) => Amount } getCurrentPrice
+     * @property {(seat: ZCFSeat) => string} addLiquidity
+     * @property {(seat: ZCFSeat) => string} removeLiquidity
+     * @property {() => ZCFSeat} getPoolSeat
+     * @property {() => AmountMath} getAmountMath
+     * @property {() => Amount} getSecondaryAmount
+     * @property {() => Amount} getCentralAmount
+     */
+
+    /** @type {Pool} */
+    const pool = {
+      getPoolSeat: () => poolSeat,
+      getAmountMath: () => zcf.getAmountMath(secondaryBrand),
+      getCentralAmountMath: () => zcf.getAmountMath(centralBrand),
+      getCentralAmount: () =>
+        poolSeat.getAmountAllocated('Central', centralBrand),
+      getSecondaryAmount: () =>
+        poolSeat.getAmountAllocated('Secondary', secondaryBrand),
+      getCurrentPrice: (centralToSecondary, inputValue) => {
```

Comment:
I would something like an argument called `fromKeyword` with values `'Central'` or `'Secondary'`.  It would make callees easier to understand.

Reply:
Can this be renamed so it doesn't conflict with the top-level getCurrentPrice? Maybe this can be bundled with michaelfig's suggestion.